File: Assignment 2/Q1b.py
size = int(input("Enter the size of grid: "))
points = []

for m in range(size):
	for n in range(size):
		points.append([m,n])
		n += 1
	m += 1

sum = 0
pairs = size**4

tot_points = len(points)

for i in range(tot_points):
	x1,y1 = points[i]
	for j in range(tot_points):
		x2,y2 = points[j]
		sum += abs(x2 - x1) + abs(y2 - y1)

avg_dist = sum/pairs
print("The average distance between two points is",avg_dist)

File: Assignment 2/Q1b.c
#include <stdio.h>
#include<stdlib.h>

int main(){
	
	int size;
	printf("Enter the length of square grid:");
	scanf("%d", &size);

	float tot_dist = 0;
	int tot_points = size * size;
	int tot_pairs = tot_points * tot_points;

	//create a list of point coordinates

	int points[tot_points][2];
	int position = 0;

	for (int i = 0; i < size; i++){
		for (int j = 0; j < size; j++){
			points [position][0] = i;
			points [position][1] = j;
			position++;

		}
	}

	//calculate the distance
	
	for (int i = 0; i < tot_points; i++){
		int x1 = points[i][0];
		int y1 = points[i][1] ;
		for (int j = 0; j < tot_points; j++){
			int x2 = points[j][0];
			int y2 = points[j][1] ;
			tot_dist += abs(x2-x1)+abs(y2-y1);
		}
	}

	float avg_dist = tot_dist/tot_pairs;

	printf("The average distance between two points is %.4f", avg_dist);
}
/*
Output:
Enter the number of linear equidistant points:6
The average distance between two points is 1.9444
*/